examples/repo_example/succinct_hash_bin_delegations.py

```diff
@@ -0,0 +1,233 @@
+from securesystemslib.signer import SSlibSigner
+
+from tuf.api.metadata import (
+    Delegations,
+    Key,
+    Metadata,
+    SuccinctRoles,
+    TargetFile,
+    Targets,
+)
+from tuf.api.serialization.json import JSONSerializer
+
+# Succinct hash bin delegation
+# ===================
+# As mentioned above, succinct hash bin delegation has the same purpose as
+# standard hash bin delegation - distribute a large number of target files over
+# multiple delegated targets metadata roles. The consequence is a smaller
+# metadata files and thus a lower network overhead for repository-client
+# communication.
+#
+# The assignment of target files to a target's metadata is done automatically,
+# based on the byte digest of the target file name. More precisely, only a
+# prefix of the target file name byte digest is needed to assign it to the
+# correct hash bin.
+#
+# In succinct hash bin delegation all delegated roles (bins) are represented by
+# one single "succinct_roles" dictionary or when a targets file is loaded with a
+# SuccinctRoles instance from tuf/api/metadata.py
+#
+# The number of bins, name prefix for all bins and key threshold are all
+# attributes that need to be configured.
+
+# Number of bins, bit length and bin number computation
+# -----------------
+# The right number of bins depends on the expected number of target files in a
+# repository. The maximum number of bins is 2 of the power of 32.
+# For the purpose of this example we choose:
+NUMBER_OF_BINS = 32
+#
+# The bit length is the number of bits that will be used to calculate the
+# corresponding bin for each target path and it's directly related to the
+# a number of bins:
+BIT_LENGTH = int(math.log2(NUMBER_OF_BINS))
+#
+# To calculate the appropriate bin number for a particular target the path to
+# that target is signed using a cryptographic algorithm (like "sha256") which
+# guarantees uniform distribution. After that, the first BIT_LENGTH of bits from
+# the byte digest are used to calculate the bin number in hexadecimal format.
+# For more information read SuccinctRoles implementation in tuf/api/metadata.py
+
+# Delegated role (bin) name format
+# -----------------
+# Each bin has a name in the format of f"{NAME_PREFIX}-{bin_number}".
+#
+# Name prefix is the common prefix of all delegated target roles (bins) in a
+# particular SuccinctRoles object. For our example it will be:
+NAME_PREFIX = "delegated_bin"
+#
+# The suffix "bin_number" is a zero-padded hexidentical number of that
+# particular bin.
+# Zero-padded means that if NUMBER_OF_BINS is a three-digit number like 256 then
+# all bin numbers must be represented by 3 digits and if there is a bin number
+# requiring less than 3 digits an additional zeros will be padded in the
+# begging. In this example, where NUMBER_OF_BINS = 256, the first bin will have
+# a name "delegated_bin-001", the tenth bin will have a name "delegated_bin-010"
+# and the hundred bin will have the name "delegated_bin-100".
```

Comment:
Please double check this paragraph: e.g. I would expect 256 bins to fit into 2 hex digits, and the tenth bin is likely "delegated_bin-0a"?

Maybe removing most of this is fine too: I don't think the reader needs all the details here

Reply:
256 in decimal is equal to 100 in hex. 
Two hex numbers can cover all numbers between 0 and 255, but you do have a point that this probably is too many details as I am explaining what is zero-padding in the `SuccinctRoles` class itself: https://github.com/theupdateframework/python-tuf/blob/8a03abfdeb5a6b6bc892859f65425be0834893a8/tuf/api/metadata.py#L1497-L1499.
Will remove that part to make the document shorter.

Reply:
If there are 256 bins then I believe the bin numbers will be 0-255

Reply:
Yes, you are correct.
Will fix the comment in this pr directly.